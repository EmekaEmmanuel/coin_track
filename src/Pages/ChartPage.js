import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchCharts } from '../redux/Chart/chartSlice';
import Chart from '../Components/Chart';
import Footer from '../Components/Footer';
import Nav from '../Components/Nav';

function ChartPage() {
  const myCharts = useSelector((state) => state.chart.charts);
  const dispatch = useDispatch();
  const { id } = useParams();

  useEffect(() => {
    let active = true;
    (async () => {
      if (active) {
        dispatch(fetchCharts(id));
      }
    })();
    return () => {
      active = false;
    };
  }, [dispatch, id]);

  return (
    <div className="app_container">
      <Nav />
      <Chart myCharts={myCharts} />
      <Footer />
    </div>
  );
}

export default ChartPage;
