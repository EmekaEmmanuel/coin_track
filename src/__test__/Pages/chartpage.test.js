import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import ChartPage from '../../Pages/ChartPage';

it('Chart page renders correctly', () => {
  const myCharts = store.getState().chart.charts;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <ChartPage myCharts={myCharts} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
