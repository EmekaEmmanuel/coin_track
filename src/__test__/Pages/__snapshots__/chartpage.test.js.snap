// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart page renders correctly 1`] = `
<div
  className="app_container"
>
  <div
    className="app_container"
  >
    <header
      className="home_header_box header_line"
    >
      <div
        className="home_header "
      >
        <div
          className="logo_box"
        >
          <figure>
            <img
              alt=""
              className="logo_img"
              src="crypto_image.png"
            />
          </figure>
          <h2
            className="nav_font1"
          >
            monie track
          </h2>
        </div>
        <nav
          className="nav_tag"
        >
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Track 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              aria-current="page"
              className="nav_font2 td_none active"
              href="/"
              onClick={[Function]}
            >
              Happenings 
            </a>
          </li>
          <li
            className="active nav_item"
          >
            <a
              activeClassName="active"
              className="nav_font2 td_none"
              href="/ChartPage"
              onClick={[Function]}
            >
              Chart 
            </a>
          </li>
        </nav>
        <div
          className="notify_box"
        >
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
          <button
            type="button"
          >
             
            <img
              alt=""
              src="closepopup.png"
            />
          </button>
        </div>
        <div
          className="hamburger_container"
        >
          <figure>
            <img
              alt="Hamburger_menu"
              src="hamburger.png"
            />
          </figure>
        </div>
      </div>
    </header>
  </div>
  <div
    className="app_container"
  >
    <main
      className="chart_main_tag"
    >
      <section
        className="chart_section1"
      >
        <p
          className="chart_font1"
        >
          LIVE MARKET CHANGES
        </p>
        <h1
          className="chart_font2"
        >
          Get an Overview on the Live chart
        </h1>
        <p
          className="chart_font3"
        >
          Your favorite place to see moment changes.
        </p>
      </section>
      <section
        className="chart_section2"
      >
        <article
          className="chart_article"
        >
          <h3
            className="chart_font4"
          >
            Live chart
          </h3>
          <div>
            <canvas
              height={400}
              role="img"
              width={300}
            />
          </div>
          <div
            className="button_box"
          >
            <button
              className="chart_section3_btn"
              type="button"
            >
              1 month
            </button>
          </div>
        </article>
      </section>
    </main>
  </div>
  <div
    className="app_container"
  >
    <footer
      className="home_footer"
    >
      <article
        className="home_footer_article"
      >
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Company
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Trade
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Help
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
        <ul>
          <p
            className="home_footer_paragraph"
          >
            Products
          </p>
          <li
            className="home_footer_list"
          >
            About
          </li>
          <li
            className="home_footer_list"
          >
            Terms of Use
          </li>
          <li
            className="home_footer_list"
          >
            Careers
          </li>
          <li
            className="home_footer_list"
          >
            Privacy Policy
          </li>
        </ul>
      </article>
      <div
        className="footer_copyright"
      >
        Â© 2022 monie track. All rights reserved.
      </div>
    </footer>
  </div>
</div>
`;
